src/app/directives/strongRoute/strong-route-active.directive.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ChangeDetectorRef,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  Input,
+  Optional,
+  QueryList,
+  Renderer2,
+} from "@angular/core";
+import {
+  IsActiveMatchOptions,
+  Router,
+  RouterLink,
+  RouterLinkActive,
+} from "@angular/router";
+import { StrongRouteDirective } from "./strong-route.directive";
+
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: "a[strongRouteActive]",
+})
+export class StrongRouteActiveDirective extends RouterLinkActive {
+  @ContentChildren(StrongRouteDirective, { descendants: true })
+  public linksWithHrefs!: QueryList<StrongRouteDirective>;
```

Comment:
is the bang (`!`) significant ?

Reply:
Somewhat, its typescript syntax thats required if i ever want to enable `strict` type checking for the project. It basically just means that this value will exist eventually. Otherwise in strict mode, it is treated as `QueryList | undefined`